src/bootstrap/src/bin/rustc.rs

```diff
@@ -91,6 +91,19 @@ fn main() {
         rustc_real
     };
 
+    // Get the name of the crate we're compiling, if any.
+    let crate_name = arg("--crate-name");
+
+    // We want everything statically linked into `rustc_driver`, so remove `-C prefer-dynamic`
```

Comment:
it could also be `-Cprefer-dynamic` without a space

Reply:
This argument parsing is quite hacky already, maybe I could just make it panic if not found?

Reply:
sgtm

Reply:
It's apparently not passed for the test build, not sure how that works, but I changed it to look for `-Cprefer-dynamic` too.